proposals/nnnn-app-services.md

```diff
@@ -0,0 +1,629 @@
+	
+</function>
+
+<function name="GetAppServiceData" functionID="GetAppServiceDataID" messagetype="response">
+	<description> This response includes the data that is requested from the specific service</description>
+	
+	<param name="serviceData" type="AppServiceData" mandatory="true"/>
+		
+</function>
+```
+
+##### Subscribed services
+
+If the app service consumer subscribed to updates from the app service provider they will recieve those updates through an `OnAppServiceData ` notification. This notification will simply contain an `AppServiceData` struct with the updated data.
+
+###### OnAppServiceData
+
+```xml
+<function name="OnAppServiceData" functionID="OnAppServiceDataID" messagetype="notification">
+	<description> This notification includes the data that is updated from the specific service</description>
+	
+	<param name="serviceData" type="AppServiceData" mandatory="true"/>
+	
+</function>
+```
+
+#### App Service Media
+
+One of the aspects to publishing services will be to include data beyond simple text strings. For our observed use cases this will likely be simple icons/small images, for example, app icons, weather icons, or album art.  This will require the app service provider to push the files to the module. This is done through the normal PutFile means.
+
+![App Service Media](../assets/proposals/NNNN-app-services/file_transfer.png "App Service Media")
+
+
+When an app service consumer receives an `AppServiceData` object that contains file names of the media that the app service provider intends to share they will need to send a `GetFile` request. The structure of the `GetFile` request/response pair is very similar to that of the `PutFile` pair.
+
+```xml
+<function name="GetFile" functionID="GetFileID" messagetype="request">
+	<description> This request is sent to the module to retrieve a file</description>
+
+	<param name="syncFileName" type="String" maxlength="255" mandatory="true"/>
+	<param name="appServiceId" type="String" mandatory="false"/>
+
+	<param name="fileType" type="FileType" mandatory="false">
+		<description>Selected file type.</description>
+	</param>
+	
+	
+	<param name="offset" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="false">
+		<description>Optional offset in bytes for resuming partial data chunks</description>
+	</param>
+	
+	<param name="length" type="Integer" minvalue="0" maxvalue="2000000000" mandatory="false">
+		<description>Optional length in bytes for resuming partial data chunks if offset is set to 0, then length is the total length of the file to be downloaded</description>
+	</param>
+		
+</function>
+
+<function name="GetFile" functionID="GetFileID" messagetype="response">
+	<description> This response includes the data that is requested from the specific service</description>
+	
+	<param name="success" type="Boolean" platform="documentation" mandatory="true">
+		<description> true if successful; false, if failed </description>
```

Comment:
Should there be a comma after "true" or should the comma after "false" be removed so that these match?